docs/stakeholder_needs/product_manufacturer.sdoc

```diff
@@ -0,0 +1,343 @@
+<<<
+
+[USER_NEED]
+UID: ZEP-143
+STATUS: Draft
+TYPE: User Need
+USER: Product Manufacturer
+TITLE: Control Power Mode of MCU and Peripherals
+STATEMENT: >>>
+As a Product Manufacturer,
+
+I need to be able to control the power mode of the MCU and its peripherals to take advantage of the hardware features and to be able to implement low power or battery driven long life applications.
+<<<
+
+[USER_NEED]
+UID: ZEP-144
+STATUS: Draft
+TYPE: User Need
+USER: Product Manufacturer
+TITLE: Runnable on MCUs/CPUs with >= 1 core
+STATEMENT: >>>
+As a Product Manufacturer,
+
+I need an OS that runs on MCUs/CPUs with one or more CPU cores.
+<<<
+
+[USER_NEED]
+UID: ZEP-145
+STATUS: Draft
+TYPE: User Need
+USER: Product Manufacturer
+TITLE: Control which Thread runs on which CPU Core
+STATEMENT: >>>
+As a Product Manufacturer,
+
+I need to be able to control which thread will run on which CPU.
+<<<
+
+[USER_NEED]
+UID: ZEP-146
+STATUS: Draft
+TYPE: User Need
+USER: Product Manufacturer
+TITLE: Thread-safe Means to Exchange Data Between Threads
+STATEMENT: >>>
+As a Product Manufacturer,
+
+I need to able to exchange information between threads in a thread-safe manner guaranteeing data consistence.
+<<<
+
+[USER_NEED]
+UID: ZEP-147
+STATUS: Draft
+TYPE: User Need
+USER: Product Manufacturer
+TITLE: Support for Threads
+STATEMENT: >>>
+As a Product Manufacturer,
+
+I need an OS that supports threads.
+<<<
+
+[USER_NEED]
+UID: ZEP-148
+STATUS: Draft
+TYPE: User Needs
+USER: Product Manufacturer
+TITLE: Management of Prioritized Threads
+STATEMENT: >>>
+As a Product Manufacturer,
+
+I need to be able to manage the execute of multiple threads with different priorities.
+<<<
+
+[USER_NEED]
+UID: ZEP-149
+STATUS: Draft
+TYPE: User Need
+USER: Product Manufacturer
+TITLE: Threads Prioritized
+STATEMENT: >>>
+As a Product Manufacturer,
+
+I need to be able to give my threads different priorities for execution.
+<<<
+
+[USER_NEED]
+UID: ZEP-150
+STATUS: Draft
+TYPE: User Need
+USER: Product Manufacturer
+TITLE: Timer Management
+STATEMENT: >>>
+As a Product Manufacturer,
+
+I need an OS that exposes controls to start, suspend, resume and stop timers.
+<<<
+
+[USER_NEED]
+UID: ZEP-151
+STATUS: Draft
+TYPE: User Need
+USER: Product Manufacturer
+TITLE: Timer Expiration Scheduling
+STATEMENT: >>>
+As a Product Manufacturer,
+
+I need an OS that exposes control to set the expiration time of a timer.
+<<<
+
+[USER_NEED]
+UID: ZEP-152
+STATUS: Draft
+TYPE: User Need
+USER: Product Manufacturer
+TITLE: Timer Expiration Event
+STATEMENT: >>>
+As a Product Manufacturer,
+
+I need an OS that triggers a event notifying the expiration of a timer.
+<<<
+
+[USER_NEED]
+UID: ZEP-153
+STATUS: Draft
+TYPE: User Need
+USER: Product Manufacturer
+TITLE: Trace OS Operations
+STATEMENT: >>>
+As a Product Manufacturer,
+
+I need to trace different OS operations.
+<<<
```

Comment:
This is quite long for a flat list. It needs some structure to be managable over time.